packages/foam-vscode/src/features/hover-provider.spec.ts

```diff
@@ -75,18 +75,18 @@ describe('Hover provider', () => {
     const { doc } = await showInEditor(noteA.uri);
     const pos = new vscode.Position(0, 22); // Set cursor position on the wikilink.
 
-    const result: vscode.Hover = provider.provideHover(
+    const promiseResult = provider.provideHover(
       doc,
       pos,
       noCancelToken
-    ) as vscode.Hover;
-
-    expect(result.contents.length).toEqual(1);
-    const mdResult = result.contents[0] as vscode.MarkdownString;
-    expect(mdResult.value).toEqual(
-      '\n```markdown\n' + fileB.content + '\n```\n'
-    );
-    expect(mdResult.isTrusted).toBeFalsy();
+    ) as Promise<vscode.Hover>;
+
+    return promiseResult.then(result => {
+      expect(result.contents).toBeDefined();
+      expect(result.contents).toEqual(
```

Comment:
This test don't work we received always an empty string!
While by doing manual tests it work as expected.

I don't understand why? Maybe the fixtures are badly initialized with the promises...
(I will check an other moment)

Reply:
Resolved by https://github.com/foambubble/foam/pull/728#discussion_r676728842